### PR TITLE
DartCode improvements

### PR DESCRIPTION
**Pull request by JonasWanke**

### CODE REVIEW

File: packages/dart_code/src/nodes.candy [hunk 1]
```diff
@@ -156,13 +190,44 @@ impl DartBlock: DartBody
...
+static fun dartIterator(entry: DartType): DartType {
+  DartNamedType(DartIdentifier("Iterator", Some<String>(dartCoreUrl)), List.of1<DartType>(entry))
+}
+static fun dartList(entry: DartType): DartType {
+  DartNamedType(DartIdentifier("List", Some<String>(dartCoreUrl)), List.of1<DartType>(entry))
+}
+static fun dartMap(key: DartType, value: DartType): DartType {
+  DartNamedType(DartIdentifier("Map", Some<String>(dartCoreUrl)), List.of2<DartType>(key, value))
+}
+static let dartNull = DartNamedType(DartIdentifier("Null", Some<String>(dartCoreUrl)), List.empty<DartType>())
```
Comment (MarcelGarus) [inline packages/dart_code/src/nodes.candy:216]: We should make sure this doesn't get confused with the `null` constant (for which we should probably also have a constant here.

**Review by MarcelGarus — APPROVED** [review]

File: packages/dart_code/src/nodes.candy (hunk 1 above)
Comment (JonasWanke) [reply packages/dart_code/src/nodes.candy:216]: For `null`, there's `DartNullLiteral()` already. Should there be a constant for that too? I'm not sure that's worth it, unless we make `DartNullLiteral` private.

Regarding the type constant: Should I rename it to `dartNullType`? That would differentiate it from other type constants, but it's clearer that it's a type. (This becomes irrelevant as soon as we can have static properties in traits, as we can move these to `DartType`.)

Comment (MarcelGarus) [reply packages/dart_code/src/nodes.candy:216]: Ah, didn't spot the `DartNullLiteral`. Then I'd leave this as-is, keep it consistent.

**Comment by MarcelGarus** [comment]
I believe this can be merged now?

**Pull request merged**